### DISCUSSION TITLE
Next js app fail to compile when deployed to vercel

### DISCUSSION THREAD

**Discussion by Chandler-Zhu** [vercel/next.js#40330] [Help] [answered]
My app works perfectly in localhost, but when I deploy it on vercel it fails to compile, error message:
![image](https://user-images.githubusercontent.com/61914365/188976506-6cbff27a-9985-4242-9084-ebb531153552.png)
my `index.tsx` :  https://github.com/Chandler-Zhu/image-gallery/blob/main/pages/index.tsx

**Comment by icyJoseph** [comment, marked as answer]
Hi,

Your repository fails locally for me too, `next build` runs a type-check which you might not be seeing locally because, typically, we use `next dev` when developing and that one ignores type-errors.

I think there's something weird about how shoelace components are exported from the root of their react package. Changing to:

```tsx
const SlButton = dynamic(
  () => import('@shoelace-style/shoelace/dist/react/button'),
  {
    ssr: false,
  }
)
const SlIcon = dynamic(
  () => import('@shoelace-style/shoelace/dist/react/icon'),
  {
    ssr: false,
  }
)
```

Does the trick. Notice that I import the actual component directly, instead of loading the entire set of components, and then load the one I care about. I think overall, this ought to be better for performance as well, as it means downloading less JavaScript.

**Reply by Chandler-Zhu** [reply]
Your solution looks cleaner too, thank you!
